content/source/docs/enterprise-beta/api/modules.html.md

```diff
@@ -0,0 +1,189 @@
+---
+layout: enterprise2
+page_title: "Modules - API Docs - Terraform Enterprise Beta"
+sidebar_current: "docs-enterprise2-api-modules"
+---
+
+# Registry Modules API
+
+-> **Note**: These API endpoints are in beta and are subject to change.
+
+## List modules
+
+Lists all the modules for a given organization. **Note** that all other endpoints have the subpath `/api/v2`, this endpoint does not use that subpath.
+
+| Method | Path           |
+| :----- | :------------- |
+| POST | /api/registry/v1/modules/:organization |
+
+
+### Parameters
+- `:organization` (`string: <required>`) - Specifies the organization for which to list the modules
+
+### Sample Request
+
+```shell
+curl \
+  --header "Authorization: Bearer $ATLAS_TOKEN" \
+  https://atlas.hashicorp.com/api/registry/v1/modules/skierkowski-v2
+```
+
+### Sample Response
+
+```json
+{
+  "meta": {
+    "limit": 15,
+    "current_offset": 0
+  },
+  "modules": [
+    {
+      "id": "skierkowski-v2/instance/aws/0.0.08",
+      "owner": "",
+      "namespace": "skierkowski-v2",
+      "name": "instance",
+      "version": "0.0.08",
+      "provider": "aws",
+      "description": "",
+      "source": "https://bitbucket-server.hashicorp.engineering/projects/SKI/repos/terraform-aws-instance/browse",
+      "published_at": "2017-11-30T07:00:40.664036Z",
+      "downloads": 0,
+      "verified": false
+    },
+    {
+      "id": "skierkowski-v2/vpc/aws/0.0.1",
+      "owner": "",
+      "namespace": "skierkowski-v2",
+      "name": "vpc",
+      "version": "0.0.1",
+      "provider": "aws",
+      "description": "",
+      "source": "https://bitbucket-server.hashicorp.engineering/projects/SKI/repos/terraform-aws-vpc/browse",
+      "published_at": "2017-11-30T00:53:06.966753Z",
+      "downloads": 0,
+      "verified": false
+    },
+    {
+      "id": "skierkowski-v2/subnet/aws/0.0.1",
+      "owner": "",
+      "namespace": "skierkowski-v2",
+      "name": "subnet",
+      "version": "0.0.1",
+      "provider": "aws",
+      "description": "",
+      "source": "https://bitbucket-server.hashicorp.engineering/projects/SKI/repos/terraform-aws-subnet/browse",
+      "published_at": "2017-11-30T17:38:01.600527Z",
+      "downloads": 0,
+      "verified": false
+    }
+  ]
+}
+```
+
+## Publish a Module
+
+This endpoint can be used to publish a new module to the registry. The publishing process will ingress the configuration from the repository and parse the various fields (e.g. input variables, output variables, readme) from the repository.
+
+| Method | Path           |
+| :----- | :------------- |
+| POST | /registry-modules |
+
+### Parameters
+
+- `linkable-repo-id` (`string: <required>`) - Specifies the repository to be used to ingress the configuration, this is in the format `<PROJECT_KEY>/<REPO>`
```

Comment:
Should we make that more VCS agnostic? For example when we have GitHub support `<PROJECT_KEY>` is a bit confusing since it's presumably just an organisation name in GitHub.

Fair enough for now if the plan is to update it later when other VCS are supported. If we use that terminology globally elsewhere then I guess fine too. It's probably obvious anyway to most devs.

Reply:
I'd cut it off at the comma and then make sentence 2:

> For BitBucket server, the format is `<PROJECT_KEY>/<REPO>`. This is the only supported VCS right now.